flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/rank/UpdatableTopNFunction.java

```diff
@@ -312,6 +355,46 @@ private void processElementWithRowNumber(RowData inputRow, Collector<RowData> ou
                 "Failed to find the sortKey, rowkey in the buffer. This should never happen");
     }
 
+    private void emitRecordsWithRowNumberIgnoreStateError(
+            RowData newSortKey,
+            RowData newRow,
+            int newRank,
+            RowData oldSortKey,
+            RankRow oldRow,
+            int oldRank,
+            Collector<RowData> out) {
+        Iterator<Map.Entry<RowData, Collection<RowData>>> iterator = buffer.entrySet().iterator();
+        int currentRank = 0;
+        RowData currentRow = null;
+        RowData prevRow = null;
+
+        while (iterator.hasNext() && currentRank <= newRank) {
+            Map.Entry<RowData, Collection<RowData>> entry = iterator.next();
+            RowData curSortKey = entry.getKey();
+            Collection<RowData> rowKeys = entry.getValue();
+            Iterator<RowData> rowKeyIter = rowKeys.iterator();
+            while (rowKeyIter.hasNext()) {
+                RowData rowKey = rowKeyIter.next();
+                currentRank += 1;
+                currentRow = rowKeyMap.get(rowKey).row;
+                if (oldRank <= currentRank) {
+                    if (currentRank == oldRank) {
+                        checkArgument(0 == sortKeyComparator.compare(curSortKey, oldSortKey));
```

Comment:
Why is this true? It might be that the `oldSortKey` is unique and after the change it does not exist anymore.

Reply:
It's a wrong check here, and  it's unnecessary after think it over, so I'll remove it.